Sources/SourceKitLSP/SourceKitLSPServer.swift

```diff
@@ -2051,9 +2065,36 @@ extension SourceKitLSPServer {
     originatorUri: DocumentURI,
     languageService: LanguageService
   ) async throws -> Location {
+    // Check if we're already in the target interface with the same module/group/symbol
+    if let referenceDoc = try? ReferenceDocumentURL(from: originatorUri),
+       case .generatedInterface(let interfaceData) = referenceDoc,
+       interfaceData.moduleName == moduleName && interfaceData.groupName == groupName {
```

Comment:
Similarly to my other comments about the build settings, we need to check here that we are using build settings from the same original file. But I think as long as we pass along the same build settings file, this shouldn’t cause any issues.